doc/evp_skey.md

```diff
@@ -0,0 +1,99 @@
+EVP_SKEY Design
+===============
+
+We currently presume that symmetric keys are represnted as octet buffer. It may
+be not relevant for the situation of hardware-backed providers, such as
+PKCS#11 or TPM.
+
+This is a design of using opaque usage in symmetric encryption.
+
+Key management
+--------------
+
+The provider operating EVP_SKEY objects has to implement a separate key
+management to deal with opaque object. We already have MAC key management and
+it looks suitable for dealing with EVP_SKEY objects.
+
+Allocation and freeing
+--------
+
+```C
+EVP_SKEY * EVP_SKEY_new(void);
+void EVP_SKEY_free(EVP_SKEY *key);
+```
```

Comment:
How do I associate an `EVP_SKEY` with a key type?  In other words, if I want an AES key, how do I ensure I will get an AES key?

Reply:
http://docs.oasis-open.org/pkcs11/pkcs11-curr/v2.40/cs01/pkcs11-curr-v2.40-cs01.html#_Toc399398916 - so yes, you are correct, the SKEYs are algorithm-aware.

Reply:
But do we care - it could be a provider-specific matter. What's important is that the EVP_SKEY is internally held inside the provider.

I am not sure whether we want `EVP_SKEY_new(void)` though. An EVP_SKEY should be created either via an import (fromdata() call), by a derive or unwrap operation, or by a reference obtained from an OSSL_STORE. Why would we want to create an _empty_ EVP_SKEY object is unclear to me.

Reply:
I think we have lots of reasons, at this point, to *not* model EVP_SKEY on EVP_PKEY.  Better think a bit more from scratch

Reply:
Previous comment scratched...  I've realized that some internal things will have to be modelled after EVP_PKEY